### PR TITLE
Update question terminology, make validation text more concise

### PR DESCRIPTION
**Pull request by drewoldag**
We received some feedback, recorded in issue #182 that the term "module" in the copier questions wasn't precise enough, it was suggested that we change that to "package", which is more accurate. The vast majority of the files that were touched are simply propagating the change from `module_name` to `package_name` through all the templates. 

It was also noted that for 80 character terminal windows the validation error text will be cut off, and doesn't scroll or wrap lines. I took a few swings at trying to get a carriage return or line break to render, but due to the way that the error is rendered, it is only displayed on a single line. (including a line break ,`<br>` simply renders _that string_ in the validation error message 🤷 ) 

So I've tried to make the error message super concise to fit under the 80 character limit and I added more information about the regex validation in the "Starting a new project" RTD question table.

### CODE REVIEW

File: docs/source/new_project.rst [hunk 1]
```diff
@@ -25,9 +25,9 @@ questions:
    * - *Would you like to use simple (default tooling) or customized installation?*
      - If a simple install is used, the template automatically selects the recommended tooling options (linter, isort, and create example module). 
    * - *What is the name of your project?*
-     - The name of your project.
-   * - *What is your python module name?*
-     - The name of your (first) module. This controls where your source code will live (``src/{{module_name}}``).
+     - The name of your project. If you distribute your code via PyPI, this is the name that will be used. This will allow users to pip install like so: ``pip install <project_name>``. The project name must start with a lowercase letter, followed by one or more of the following (a-z, 0-9, _, -).
+   * - *What is your python package name?*
+     - The name of your (first) module. This controls where your source code will live (``src/{{package_name}}``).  The package name must start with a lowercase letter, followed by one or more of the following (a-z, 0-9, _).
```
Comment (smcguire-cmu) [inline docs/source/new_project.rst:30]: I think that this first sentence should read `The name of your top level package.` If I understand correctly, a module in python is a `.py` file, and a package is a folder with an `__init__.py` https://docs.python.org/3/tutorial/modules.html#packages

File: docs/source/new_project.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ questions:
    * - *Would you like to include mypy to perform static type checking for type hints?*
      - `mypy <https://www.mypy-lang.org>`_ performs static type checking on python code that uses `type hints <https://docs.python.org/3/library/typing.html>`_. This type checking makes sure that the correct data types are being used where type hints are defined. If basic or strict type checking is selected, a pre-commit hook and GitHub actions workflow that perform the type checking are added. Basic type checking performs type checks but ignores code or imports for which type hints are not written. Strict type checking enforces type hints are used by giving errors where no type hints are found.
    * - *Do you want to create some example module code?*
-     - If this option is selected the template will create a model in ``src/{{module_name}}`` and create a corresponding example test file. Defaults to ``True`` during simple installation.
+     - If this option is selected the template will create a module in ``src/{{package_name}}`` and create a corresponding example test file. Defaults to ``True`` during simple installation.
```
Comment (smcguire-cmu) [inline docs/source/new_project.rst:44]: Not sure if this shouldn't be too long, but could we include the name of the module in the example path here? i.e. `src/{{package_name}}/example_module.py` (I think that's the name of the example module we use)

**Review by smcguire-cmu — COMMENTED** [review]
Thanks for making the changes! Just a few nit picks but mostly looks good

File: docs/source/new_project.rst (hunk 1 above)
Comment (drewoldag) [reply docs/source/new_project.rst:30]: Fixed.

File: docs/source/new_project.rst (hunk 2 above)
Comment (drewoldag) [reply docs/source/new_project.rst:44]: Updated

**Review by smcguire-cmu — APPROVED** [review]
Lgtm, thanks!

**Pull request merged**